design/one-pager-consuming-k8s-clusters.md

```diff
@@ -0,0 +1,321 @@
+executed by [almost disjoint] sets of users. Having these actions represented by
+separate custom resources allows for more granular [RBAC] permissions around
+Kubernetes cluster usage in Crossplane.
+
+### Automatic KubernetesTarget Creation Controller
+
+Because the ability to *publish* a Kubernetes cluster provides immense power in
+a Crossplane environment, it is desirable for Kubernetes cluster managed
+resources to be able to be consumed in the namespace that they are claimed
+without a user requiring the ability to create a `KubernetesTarget`. For
+instance, if an infrastructure owner wants to allow for an application team to
+provision some clusters dynamically by creating a `KubernetesCluster` claim in
+their namespace that references a `GKEClusterClass`, that application team
+should be able to then schedule `KubernetesApplication` resources to that
+cluster without requiring the permissions to create a `KubernetesTarget` (which
+would allow them to enable to consumption of any Kubernetes cluster managed
+resource) or asking the infrastructure owner to create a `KubernetesTarget` for
+them (which would break the self-service model).
+
+To enable this functionality, a single controller should be implemented in core
+Crossplane that watches for the creation of `KubernetesCluster` claim resources
+and automatically creates a `KubernetesTarget` with the `connectionSecretRef`
+set to the connection `Secret` that was propagated to the namespace.
+Importantly, this controller should also clean up the `KubernetesTarget` when
+the `KubernetesCluster` (and corresponding) `Secret` are deleted. This can be
+accomplished by setting an `ownerReference` to the `KubernetesCluster` claim on
+the `KubernetesTarget`. In addition, because this `KubernetesTarget` resource
+may be scheduled to by a `KubernetesApplication` using labels, any labels that
+are present on the `KubernetesCluster` claim should be propagated to the
+`KubernetesTarget` resource and the name of the resource should match that of
+the claim.
+
+### User Workflows
+
+The following scenarios serve to illustrate possible workflows that would be
+enabled by this model. While some of these scenarios appear less likely than
+others, the primary takeaway is the flexibility of this model, which allows for
+a variety of use-cases.
+
+#### Scenario 1: Infrastructure owner provisions and publishes all Kubernetes clusters
+
+1. Infrastructure owner statically provisions three `GKECluster` managed
+   resources: `dev`, `stage`, and `prod`
+1. An application owner asks for an environment to deploy their application.
+1. Infrastructure owner creates a new namespace and a `KubernetesTarget` that
+   points to the `dev` cluster.
+1. Application owner provisions a `KubernetesApplication` in that namespace with
+   labels that select the `KubernetesTarget` referencing the `dev` cluster for
+   provisioning.
+1. When ready for testing, the application owner requests access to the `stage`
```

Comment:
The simplified UX of cluster-scoped classes push things towards a crossplane instance per environment, and while you can definitely model classes with environment labels (dev, stage, prod), there is no means of securing access via RBAC via labels for resource class usage AFAIK, so wonder if a cluster-scoped `KubernetesTarget` would be a simpler way to go.